Dev bran controller

B08waffles opened the pull request:
Added two features and a test for each. 
CORS has been added server wide on app.ts, going with default settings for now with discussions on whether we go
server wide or route specific for certain differences in settings? 
Rate limits in the form of "express-rate-limit" has been added to app.ts as well, here the rate is set to allow up to 2 request per second, a number that I am happy to change if need be. 

I also added the --verbose flag to the test script for slightly more detail when debugging failed tests.

Commit:
Feat: Add CORS application wide

Commit:
Fix: Remove unused variables

Commit:
Feat: Add 2/second rate limit

Commit:
Fix: stage and commit migrations

Commit:
Fix: Ran `npm run ci` to fix formatting

2016lisali commented on __test__/integrationTests.test.ts:
```diff
@@ -12,28 +12,46 @@ let expressInstance: Express;
+    // to avoid the test application from leaking memory and allows Jest to exit
+    // cleanly.
+    store = new PrismaSessionStore(prisma, {
+        checkPeriod: 10 * 60 * 1000, // 1 hour in milliseconds
```
should it be 60 *60 *1000 = 1 hour?

Jet004 replied:
You are correct. This is code I wrote in a previous pull request. Are you happy for me to change it in my next update and push it with the logout code? Good job noticing this!

2016lisali replied:
Yep :)

Jet004 replied:
This update has already been merged to the development branch 😄

2016lisali commented on index.ts:
```diff
@@ -9,11 +9,11 @@ const PORT: string | undefined = process.env.PORT || "3000";
 const store = new PrismaSessionStore(prisma, {
-	checkPeriod: 10 * 60 * 1000, // 1 hour in milliseconds
-	dbRecordIdIsSessionId: true,
+    checkPeriod: 10 * 60 * 1000, // 1 hour in milliseconds
```
Should it be 60 * 60 * 1000?

Jet004 replied:
Same as above comment

Jet004 commented on app.ts:
```diff
@@ -2,36 +2,52 @@ import express, { Express } from "express";
+        legacyHeaders: false, // Disable the `X-RateLimit-*` headers
+    });
+    server.use(limiter);
+
```
The one thing I would suggest is to add the CORS and rate limit middleware before the session middleware. This way if a request is blocked by either CORS or rate limiting, the session middleware won't run --> this reduces (very slightly) server computations and DB queries. More efficient and reduces server load if there are a lot of requests.

B08waffles replied:
Good idea! Now that you point it out it makes a lot more sense then leaving it like this, i'll readjust this in the next commit. Thanks for your attention to detail.

2016lisali reviewed (approved):
Looks Good :)

Commit:
Fix: Move CORS & Rate Limits before Session

Commit:
Fix: Add "/.log" to .gitignore

Jet004 committed:
fix: pull latest changes from development and resolve conflicts

While resolving conflicts it was necessary to limit the rate limiter from running in the test environment. This is enabled only for the rate limiter tests. Tests were also modified to use axios for tests rather than supertest. This is for consistency across tests in the project.
